### PR TITLE
[SYCL][FPGA] Remove deprecation message for intel::disable_loop_pipelining attribute

### PR DESCRIPTION
**Pull request by smanna12**
https://github.com/intel/llvm/pull/6254 added the fpga_pipeline attribute and
deprecated the intel::disable_loop_pipelining attribute.
While this was in the pipeline, the internal decision was made to switch to properties
rather than attributes for this type of thing.

At this point, internal request is that (at a minimum) the deprecation message for
intel::disable_loop_pipelining be removed, as we no longer plan to deprecate it.
Optionally the intel::fpga_pipeline attribute can be removed from the frontend as well.

Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

**Comment by smanna12** [comment]
Failures are not related to my PR:

https://github.com/intel/llvm/runs/7214257852?check_suite_focus=true
********************
Failed Tests (1):
  SYCL :: Basic/stream/release_resources_test.cpp

https://github.com/intel/llvm/runs/7214258579?check_suite_focus=true

********************
Failed Tests (1):
  SYCL :: Assert/assert_in_simultaneously_multiple_tus.cpp

**Review by mendell27 — DISMISSED** [review]

**Comment by premanandrao** [comment]
> Optionally the intel::fpga_pipeline attribute can be removed from the frontend as well.

I am assuming this will be done in a later patch?

File: clang/test/SemaSYCL/intel-fpga-loops.cpp [hunk 1]
```diff
@@ -269,6 +272,10 @@ void zoo() {
   [[intel::max_concurrency(2)]]
   [[intel::initiation_interval(2)]] for (int i = 0; i != 10; ++i)
       a[i] = 0;
+  [[intel::disable_loop_pipelining]]
+  // expected-error@+1 {{duplicate Intel FPGA loop attribute 'disable_loop_pipelining'}}
+  [[intel::disable_loop_pipelining]] for (int i = 0; i != 10; ++i)
+    a[i] = 0;
   [[intel::fpga_pipeline]]
```
[the remark this answers is not in the document]
Comment (smanna12) [reply clang/test/SemaSYCL/intel-fpga-loops.cpp:279]: Done

**Comment by smanna12** [comment]
> > Optionally the intel::fpga_pipeline attribute can be removed from the frontend as well.
> 
> I am assuming this will be done in a later patch?

Yes, this will be done in a later patch.

**Review by premanandrao — APPROVED** [review]

**Comment by smanna12** [comment]
Thank you @premanandrao and @mendell27 for reviews.

**Comment by smanna12** [comment]
@intel/llvm-gatekeepers, this PR is ready to be merged. Thank you

**Pull request merged**